### DISCUSSION TITLE
How to intialize a CustomeNode with its width/heigth?

### DISCUSSION THREAD

**Discussion by homoroselaps** [xyflow/xyflow#3219] [Q&A] [answered]
Hello,
I want to build a custom node which size is determined by the node's fields width & height and which can be changed afterwards with the `<NodeResizer />`. I was expecting that the node starts with a size of 100 x 100 upon load but instead the style attributes width/height are only set upon first resize.
This is what happens:
![Screen Recording 2023-07-12 at 18 55 30](https://github.com/wbkd/react-flow/assets/7871498/8b4c25b8-9426-4e3e-a2b5-81e0b63ae5fd)

The css properties of the encapsulating `<div>` created by reactflow before:
<img width="419" alt="Screenshot 2023-07-12 at 18 37 56" src="https://github.com/wbkd/react-flow/assets/7871498/a8110ec4-2af6-4143-8ccc-dfa9178ea3fa">

and after:
<img width="430" alt="Screenshot 2023-07-12 at 18 38 08" src="https://github.com/wbkd/react-flow/assets/7871498/4a497d47-5d5e-442a-a636-fd83cf5e1939">

When loading my app I load this node:
`{
  "data": {},
  "position": {
    "x": 241.5,
    "y": 243,
    "vx": 0,
    "vy": 0
  },
  "className": "",
  "style": {},
  "targetPosition": "left",
  "hidden": false,
  "selected": false,
  "expandParent": false,
  "selectable": true,
  "connectable": true,
  "dragging": false,
  "draggable": true,
  "deletable": true,
  "sourcePosition": "right",
  "width": 100,
  "height": 100,
  "zIndex": 10,
  "type": "repo",
  "label": "repo",
  "id": "025b413f8a"
}`

This is the code for my customNode:

```import { NodeProps } from 'reactflow';
import { NodeResizer } from '@reactflow/node-resizer';
interface GroupNodeProps { nodeProps: NodeProps; }

function TestNode({ nodeProps: { id, data: { title, subline } } }: GroupNodeProps) {
    return (
        <>
            <p className="text-white">{title}</p>
            <p className="text-white">{subline}</p>
            <NodeResizer />
        </>
    );
}

export default TestNode;```

**Comment by bcakmakoglu** [comment, marked as answer]
Have you tried setting to `node.style = { width: 100, height: 100 }`?

**Reply by homoroselaps** [reply]
Yes this does the job. I kinda expected that a manual style is given higher priority and hence causing the node to be static at this size.
Setting the style in addition to width/height is inconvenient. Do you think this might be considered a bug in reactflow?

**Reply by bcakmakoglu** [reply]
I don't think this is a bug, strictly speaking.
The documentation might not be fully clear on this but as far as I can tell `node.width` and `node.height` are not meant to be set to initialize any node size but instead are values that are written as somewhat of a "readonly" prop after they have been read by a ResizeObserver.

So they aren't even applied as a style or anything, it's just the result of some measurements that are necessary for other parts of the library to work.

**Reply by bcakmakoglu** [reply]
[Also from what I can tell they aren't even documented as node options on the docs.](https://reactflow.dev/docs/api/nodes/node-options/)

Though the TypeScript interface for `Node` suggests that you *could* pass `node.width` and `node.height` without indicating that it's going to do nothing to actually change the nodes size.

**Reply by homoroselaps** [reply]
What you say seems plausible. There doesn't seem to exist a standard way to initializing a NodeResizer with width / height (besides setting the styles) similar to the minWidth & minHeight property... Or do I miss something here?

**Reply by moklick** [reply]
yep. It uses the node dimensions which can be defined by using style.width/ height as said above.
